nixos/modules/services/web-apps/gerrit.nix

```diff
@@ -0,0 +1,218 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.gerrit;
+
+  # NixOS option type for git-like configs
+  gitIniType = with types;
+    let
+      primitiveType = either str (either bool int);
+      multipleType = either primitiveType (listOf primitiveType);
+      sectionType = lazyAttrsOf multipleType;
+      supersectionType = lazyAttrsOf (either multipleType sectionType);
+    in lazyAttrsOf supersectionType;
```

Comment:
`lazyAttrsOf` should ideally only be used if either `mkIf false` definitions don't make sense (which isn't the case here), or if such definitions are handled with an `emptyValue` that then gets filtered out (e.g. by using `lazyAttrsOf (nullOr ...)`, then filtering out `null`'s before generating the file). Because this isn't done here, the following will error:
```nix
{
  services.gerrit.settings.foo.bar = mkIf false 10;
}
```
```
error: The option `services.gerrit.settings.foo.bar' is used but not defined.
```

Reply:
Or alternatively just using `attrsOf` instead

Reply:
So just replace all the `lazyAttrsOf` with `attrsOf`? I must admit that I cargo-culted this and don't have a good understanding of these distinctions even after your explanation.